app/src/main/java/com/github/gotify/api/CertUtils.java

```diff
@@ -69,7 +74,43 @@ public static void applySslSettings(OkHttpClient.Builder builder, SSLSettings se
                             context.getSocketFactory(), (X509TrustManager) trustManagers[0]);
                 }
             }
-        } catch (Exception e) {
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                if (settings.clientCert != null) {
+                    KeyStore ks = KeyStore.getInstance("PKCS12");
+                    InputStream bs = new ByteArrayInputStream(Base64.getDecoder().decode(settings.clientCert));
+                    ks.load(bs, settings.clientCertPassword.toCharArray());
+
+                    KeyManagerFactory kmf = KeyManagerFactory.getInstance("X509");
+                    kmf.init(ks, settings.clientCertPassword.toCharArray());
+
+
+                    TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+                            TrustManagerFactory.getDefaultAlgorithm());
+                    trustManagerFactory.init((KeyStore) null);
+                    TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+                    if (trustManagers.length != 1 || !(trustManagers[0] instanceof X509TrustManager)) {
+                        throw new IllegalStateException("Unexpected default trust managers:");
+                    }
+                    X509TrustManager trustManager = (X509TrustManager) trustManagers[0];
+
+
+                    SSLContext context = SSLContext.getInstance("TLS");
+                    context.init(kmf.getKeyManagers(), new TrustManager[] { trustManager }, null);
+                    builder.sslSocketFactory(
+                            context.getSocketFactory(), trustManager);
+                }
+            }
+        } catch (IOException iex) {
+            String tx = iex.toString();
+            if (iex.toString().contains("wrong password")) {
+                Log.e("Incorrect client certificate password.", iex);
+                return;
+            }
+
+            Log.e("Error opening client certificate.", iex);
+        }
```

Comment:
This catch can be omitted, because the exception message is already sufficient and says that the password is wrong / the certificate is invalid.